### PR TITLE
fix(mempool): change "mempool is full" log level to debug

### PR DESCRIPTION
**Pull request by yihuang**
it happens in benchmark or production when traffic is high, the log frequency is very high because it's triggered by p2p tx propagation message.

Solution:
- change it to debug level

### CODE REVIEW

**Review by melekes — APPROVED** [review]
utACK 👍

**Comment by melekes** [comment]
https://github.com/cometbft/cometbft/pull/4125

**Review by andynog — APPROVED** [review]
Usually I'd advocate to log something as an error (`logger.Error`) if it's an error that is occurring.  But I understand the impact that this might cause and affect operations. Saying that, should the log level also change on this [line](https://github.com/cometbft/cometbft/blob/dec3c8cc92b4b8e80dddce2a5467bf2541acc89e/mempool/clist_mempool.go#L367)? I'd think that would also flood the log (it's not `Info` but if `Error` level is set it might also cause the issue.

**Comment by yihuang** [comment]
> Saying that, should the log level also change on this [line](https://github.com/cometbft/cometbft/blob/dec3c8cc92b4b8e80dddce2a5467bf2541acc89e/mempool/clist_mempool.go#L367)? I'd think that would also flood the log (it's not Info but if Error level is set it might also cause the issue.

yeah, but it's less frequent since most of the requests should have been blocked at the beginning.

File: mempool/reactor.go [hunk 1]
```diff
@@ -172,12 +172,15 @@ func (memR *Reactor) TryAddTx(tx types.Tx, sender p2p.Peer) (*abcicli.ReqRes, er
...
-	case errors.Is(err, ErrTxInCache):
-		memR.Logger.Debug("Tx already exists in cache", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID)
-		return nil, err
-	case err != nil:
-		memR.Logger.Info("Could not check tx", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID, "err", err)
+	if err != nil {
+		switch {
+		case errors.Is(err, ErrTxInCache):
+			memR.Logger.Debug("Tx already exists in cache", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID)
+		case errors.Is(err, ErrMempoolIsFull{}):
```
Comment (yihuang) [inline mempool/reactor.go:179]:
```suggestion
		case errors.As(err, &ErrMempoolIsFull{}):
```

File: mempool/reactor.go [hunk 2]
```diff
@@ -172,12 +172,15 @@ func (memR *Reactor) TryAddTx(tx types.Tx, sender p2p.Peer) (*abcicli.ReqRes, er
...
-		memR.Logger.Debug("Tx already exists in cache", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID)
-		return nil, err
-	case err != nil:
-		memR.Logger.Info("Could not check tx", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID, "err", err)
+	if err != nil {
+		switch {
+		case errors.Is(err, ErrTxInCache):
+			memR.Logger.Debug("Tx already exists in cache", "tx", log.NewLazySprintf("%X", tx.Hash()), "sender", senderID)
+		case errors.As(err, &ErrMempoolIsFull{}):
+			memR.Logger.Debug(err.Error())
```
Comment (yihuang) [inline mempool/reactor.go:180]:
```suggestion
 			// using debug level to avoid flooding when traffic is high
			memR.Logger.Debug(err.Error())
```

**Comment by yihuang** [comment]
can we backport this to 0.38.x which we are using.
and do I need to add a changelog for this?

**Comment by melekes** [comment]
> should the log level also change on this [line](https://github.com/cometbft/cometbft/blob/dec3c8cc92b4b8e80dddce2a5467bf2541acc89e/mempool/clist_mempool.go#L367)?

I would say yes 👍

**Comment by yihuang** [comment]
> > should the log level also change on this [line](https://github.com/cometbft/cometbft/blob/dec3c8cc92b4b8e80dddce2a5467bf2541acc89e/mempool/clist_mempool.go#L367)?
> 
> I would say yes 👍

changed too.

**Review by melekes — APPROVED** [review]

**Comment by melekes** [comment]
looks like you forgot to format the Go code

**Comment by yihuang** [comment]
> looks like you forgot to format the Go code

done

**Review by andynog — APPROVED** [review]
lgtm, thanks @yihuang

**Pull request merged**